packages/angular_devkit/build_angular/src/webpack/configs/common.ts

```diff
@@ -258,6 +263,18 @@ export async function getCommonConfig(wco: WebpackConfigOptions): Promise<Config
     });
   }
 
+  if (main || polyfills) {
+    extraRules.push({
+      test: /\.[jt]sx?$/,
```

Comment:
```suggestion
      test: /\.[cm]?[tj]sx?$/,
```
That should cover all JS and upcoming TS variations.